channel/consensus_channel/consensus_channel.go

```diff
@@ -146,6 +146,18 @@ func (c *ConsensusChannel) HasRemovalBeenProposedFor(target types.Destination) b
 	return false
 }
 
+// HasRemovalBeenProposedNext returns whether or not the next proposal in the queue is a remove proposal for the given target
+func (c *ConsensusChannel) HasRemovalBeenProposedNext(target types.Destination) bool {
+
+	if len(c.proposalQueue) == 0 {
```

Comment:
I'm looking at this line https://github.com/statechannels/go-nitro/blob/b426f009f463c5e62333a9b5772e3ab53248d340/channel/consensus_channel/leader_channel.go#L24 and wondering if there is an opportunity to increase consistency by reusing `latestProposedVars()`?

Reply:
Although `latestProposedVars()` is a little difficult to follow 😟

Reply:
`latestProposedVars` is a bit more difficult for removals since we're looking for an absence of a guarantee. I think that would still work but it seems easier to just check the proposal queue.